### PR TITLE
[SD] Update SD CLI to use model_db.json

### PR DESCRIPTION
**Pull request by Shukla-Gaurav**
Signed-Off-by: Gaurav Shukla <gaurav@nod-labs.com>

### CODE REVIEW

File: shark/examples/shark_eager/dynamo_demo.ipynb [hunk 1]
```diff
@@ -36,7 +36,9 @@
     "    from torchdynamo.optimizations.backends import create_backend\n",
     "    from torchdynamo.optimizations.subgraph import SubGraph\n",
     "except ModuleNotFoundError:\n",
-    "    print(\"Please install TorchDynamo using pip install git+https://github.com/pytorch/torchdynamo\")\n",
```
Comment (qedawkins) [inline shark/examples/shark_eager/dynamo_demo.ipynb:39]: Why are these changes included in this PR? Is this coming from the linter?

File: shark/examples/shark_inference/stable_diffusion/main.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def end_profiling(device):
     neg_prompt = args.negative_prompts
     height = 512  # default height of Stable Diffusion
     width = 512  # default width of Stable Diffusion
-    if args.version == "v2.1" and args.variant == "stablediffusion":
+    if args.version == "v2_1" and args.variant == "stablediffusion":
```
Comment (qedawkins) [inline shark/examples/shark_inference/stable_diffusion/main.py:54]: My understanding was that 768x768 was a facet of sd2.1 so there is no need to check that the variant is the default as well. Correct me if I'm wrong.

File: web/models/stable_diffusion/main.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def stable_diff_inf(
     # set height and width.
     height = 512  # default height of Stable Diffusion
     width = 512  # default width of Stable Diffusion
-    if args.version == "v2_1":
+    if args.version == "v2_1" and args.variant == "stablediffusion":
```
Comment (qedawkins) [inline web/models/stable_diffusion/main.py:48]: ditto on the above.

File: web/models/stable_diffusion/stable_args.py [hunk 4]
```diff
@@ -145,6 +145,13 @@
 ### Misc. Debug and Optimization flags
 ##############################################################################
 
+p.add_argument(
+    "--use_compiled_scheduler",
+    default=True,
+    action=argparse.BooleanOptionalAction,
+    help="use the default scheduler precompiled into the model if available",
+)
```
Comment (qedawkins) [inline web/models/stable_diffusion/stable_args.py:153]: At the moment I don't think this, nor warmup count has any function. I assume that is to be added once CLI and this is merged.

**Review by qedawkins — CHANGES_REQUESTED** [review]

File: web/models/stable_diffusion/stable_args.py (hunk 4 above)
Comment (qedawkins) [reply web/models/stable_diffusion/stable_args.py:153]: (if that is the plan this comment doesn't need addressing)

File: shark/examples/shark_eager/dynamo_demo.ipynb (hunk 1 above)
Comment (Shukla-Gaurav) [reply shark/examples/shark_eager/dynamo_demo.ipynb:39]: yeah. Each time I run formatter, it pops up. So I thought to put it up. Just the formatting thing.

File: shark/examples/shark_inference/stable_diffusion/main.py (hunk 2 above)
Comment (Shukla-Gaurav) [reply shark/examples/shark_inference/stable_diffusion/main.py:54]: yeah, Should we get rid of version flag? we should have only one flag `variant`(SD1.4, SD2.1, AV3 etc).

File: shark/examples/shark_eager/dynamo_demo.ipynb (hunk 1 above)
Comment (qedawkins) [reply shark/examples/shark_eager/dynamo_demo.ipynb:39]: interesting, I don't get that. Let me see if my formatter tries to put it back.

Comment (qedawkins) [reply shark/examples/shark_eager/dynamo_demo.ipynb:39]: oh it's probably because you have this installed
```
pip install black[jupyter]
```
and no one else does. We can just leave this in then if it's annoying to take out.

File: web/models/stable_diffusion/stable_args.py (hunk 4 above)
Comment (Shukla-Gaurav) [reply web/models/stable_diffusion/stable_args.py:153]: These flags won't have any effect in the web(there are other flags though like --use_tuned that is useless right now in the web). I have placed them here just to make sure everything matches with CLI, which will make merging process easier.

File: shark/examples/shark_inference/stable_diffusion/main.py (hunk 2 above)
Comment (qedawkins) [reply shark/examples/shark_inference/stable_diffusion/main.py:54]: Is it possible to port the variant weights into a different SD version? If not then yeah, we only need variant.

File: web/models/stable_diffusion/stable_args.py (hunk 4 above)
Comment (qedawkins) [reply web/models/stable_diffusion/stable_args.py:153]: nice nice

File: shark/examples/shark_inference/stable_diffusion/main.py (hunk 2 above)
Comment (Shukla-Gaurav) [reply shark/examples/shark_inference/stable_diffusion/main.py:54]: We can check this with @PhaneeshB tomorrow.

Comment (qedawkins) [reply shark/examples/shark_inference/stable_diffusion/main.py:54]: sounds good

File: shark/examples/shark_eager/dynamo_demo.ipynb (hunk 1 above)
Comment (Shukla-Gaurav) [reply shark/examples/shark_eager/dynamo_demo.ipynb:39]: yes, black[jupyter] is there in my env.

**Review by qedawkins — APPROVED** [review]

**Review by qedawkins — APPROVED** [review]
